### PR TITLE
[ADMINAPI-979] Security Compatibility Library to .NET 8

### PR DESCRIPTION
**Pull request by JBrenesSimpat**
_The steps used in the Security.DataAccess migration was followed._

All test passed 
image

### CODE REVIEW

File: eng/postgres-docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: '3'
 services:
   db:
     container_name: Postgres-SecurityCompatibility
-    image: postgres:12-alpine
+    image: postgres:13.13-alpine3.18@sha256:3cf9bba86013877375df15001a31b10be6834e56e69ac1cb1605f469c5c5bdd6
```
Comment (JBrenesSimpat) [inline eng/postgres-docker-compose.yml:5]: Taken from a non-reviewed PR by Stephen: "The prior version is missing a crypto library"

File: .github/workflows/release-security-data-access-compatibility-53.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
         shell: pwsh
     outputs:
       hash-code: ${{ steps.hash-code.outputs.hash-code }}
-      security-data-access-version: 5.3.66
+      security-data-access-version: 5.3.67
```
Comment (JBrenesSimpat) [inline .github/workflows/release-security-data-access-compatibility-53.yml]: I saw this change in previous commits, do you know if this is the correct value ?

File: NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatiblity53.DataAccess/Models/ClaimSet.cs [hunk 3]
```diff
@@ -18,6 +18,9 @@ public class ClaimSet
         [Required]
         public string ClaimSetName { get; set; }
 
+        [Column("Application_ApplicationId")]
```
Comment (JBrenesSimpat) [inline NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatiblity53.DataAccess/Models/ClaimSet.cs:21]: NET8.0 migration makes this type of changes necessary

File: NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatiblity53.DataAccess/Contexts/PostgresSecurityContext.cs [hunk 4]
```diff
@@ -3,58 +3,65 @@
...
 
-        private class ForeignKeyLowerCaseNamingConvention : IStoreModelConvention<AssociationType>
-        {
-            public void Apply(AssociationType association, DbModel model)
-            {
-                Preconditions.ThrowIfNull(association, nameof(association));
-                Preconditions.ThrowIfNull(model, nameof(model));
+            modelBuilder.Model.GetEntityTypes().Single(e => e.ClrType.Name == nameof(ClaimSetResourceClaim))
+                .GetProperty("ClaimSetId")
+                .SetColumnName("claimset_claimsetid");
```
Comment (JBrenesSimpat) [inline NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatiblity53.DataAccess/Contexts/PostgresSecurityContext.cs:40]: NET8.0 migration makes this type of changes necessary to map this type of fields.

File: .github/workflows/release-security-data-access-compatibility-53.yml (hunk 2 above)
Comment (stephenfuqua) [reply .github/workflows/release-security-data-access-compatibility-53.yml]: Needs to match the version number in https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-Compatibility-Libraries/blob/main/NuGet/EdFi.Security.DataAccess/53/Directory.Build.props

Let's call this 5.4.0.

File: NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatibility53.DataAccess.IntegrationTests/EdFi.SecurityCompatibility53.DataAccess.IntegrationTests.csproj [hunk 5]
```diff
@@ -1,29 +1,26 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (stephenfuqua) [inline NuGet/EdFi.Security.DataAccess/53/EdFi.SecurityCompatibility53.DataAccess.IntegrationTests/EdFi.SecurityCompatibility53.DataAccess.IntegrationTests.csproj:3]: Unit test project is still using netcoreapp3.1.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**